headers.js

```diff
@@ -7,6 +7,47 @@ var ZERO_OFFSET = '0'.charCodeAt(0)
 var USTAR = 'ustar\x0000'
 var MASK = parseInt('7777', 8)
 
+// Divide this number to get high 32 bit (JS unsafe)
+var DIVIDEND_HIGH_32BIT = Math.pow(2, 32)
+
+var MAX_SAFE_INTEGER = Number.MAX_SAFE_INTEGER || (Math.pow(2, 53) - 1)
+
+// isInteger polyfill
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isInteger
+var isInteger = Number.isInteger || function (value) {
+  return typeof value === 'number' && isFinite(value) && Math.floor(value) === value
+}
+
+// isSafeInteger polyfill
+// https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Number/isSafeInteger
+var isSafeInteger = Number.isSafeInteger || function (value) {
+  return isInteger(value) && Math.abs(value) <= MAX_SAFE_INTEGER
+}
+
+var getUnisgnedHigh32 = function (val) {
```

Comment:
There is a typo in the function name

Reply:
fixed in ceb334b74dd411a2d244e99615ccba7148058154